pkg/resource/document/sdk.go

```diff
@@ -0,0 +1,1034 @@
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkFind")
+	defer func() {
+		exit(err)
+	}()
+	// If any required fields in the input shape are missing, AWS resource is
+	// not created yet. Return NotFound here to indicate to callers that the
+	// resource isn't yet created.
+	if rm.requiredFieldsMissingFromReadOneInput(r) {
+		return nil, ackerr.NotFound
+	}
+
+	input, err := rm.newDescribeRequestPayload(r)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *svcsdk.GetDocumentOutput
+	resp, err = rm.sdkapi.GetDocumentWithContext(ctx, input)
+	rm.metrics.RecordAPICall("READ_ONE", "GetDocument", err)
+	if err != nil {
+		if reqErr, ok := ackerr.AWSRequestFailure(err); ok && reqErr.StatusCode() == 404 {
+			return nil, ackerr.NotFound
+		}
+		if awsErr, ok := ackerr.AWSError(err); ok && awsErr.Code() == "InvalidDocument" {
+			return nil, ackerr.NotFound
+		}
+		return nil, err
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := r.ko.DeepCopy()
+
+	if resp.Content != nil {
+		ko.Spec.Content = resp.Content
+	} else {
+		ko.Spec.Content = nil
+	}
+	if resp.CreatedDate != nil {
+		ko.Status.CreatedDate = &metav1.Time{*resp.CreatedDate}
+	} else {
+		ko.Status.CreatedDate = nil
+	}
+	if resp.DisplayName != nil {
+		ko.Spec.DisplayName = resp.DisplayName
+	} else {
+		ko.Spec.DisplayName = nil
+	}
+	if resp.DocumentFormat != nil {
+		ko.Spec.DocumentFormat = resp.DocumentFormat
+	} else {
+		ko.Spec.DocumentFormat = nil
+	}
+	if resp.DocumentType != nil {
+		ko.Spec.DocumentType = resp.DocumentType
+	} else {
+		ko.Spec.DocumentType = nil
+	}
+	if resp.DocumentVersion != nil {
+		ko.Status.DocumentVersion = resp.DocumentVersion
+	} else {
+		ko.Status.DocumentVersion = nil
+	}
+	if resp.Name != nil {
+		ko.Spec.Name = resp.Name
+	} else {
+		ko.Spec.Name = nil
+	}
+	if resp.Requires != nil {
+		f8 := []*svcapitypes.DocumentRequires{}
+		for _, f8iter := range resp.Requires {
+			f8elem := &svcapitypes.DocumentRequires{}
+			if f8iter.Name != nil {
+				f8elem.Name = f8iter.Name
+			}
+			if f8iter.RequireType != nil {
+				f8elem.RequireType = f8iter.RequireType
+			}
+			if f8iter.Version != nil {
+				f8elem.Version = f8iter.Version
+			}
+			if f8iter.VersionName != nil {
+				f8elem.VersionName = f8iter.VersionName
+			}
+			f8 = append(f8, f8elem)
+		}
+		ko.Spec.Requires = f8
+	} else {
+		ko.Spec.Requires = nil
+	}
+	if resp.ReviewStatus != nil {
+		ko.Status.ReviewStatus = resp.ReviewStatus
+	} else {
+		ko.Status.ReviewStatus = nil
+	}
+	if resp.Status != nil {
+		ko.Status.Status = resp.Status
+	} else {
+		ko.Status.Status = nil
+	}
+	if resp.StatusInformation != nil {
+		ko.Status.StatusInformation = resp.StatusInformation
+	} else {
+		ko.Status.StatusInformation = nil
+	}
+	if resp.VersionName != nil {
+		ko.Spec.VersionName = resp.VersionName
+	} else {
+		ko.Spec.VersionName = nil
+	}
+
```

Comment:
it looks like [SSM Document have a state](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_DocumentDescription.html#systemsmanager-Type-DocumentDescription-Status), so probably we want to requeue in both sdkFind and sdkCreate until the Docment is in ACTIVE state.